include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -274,11 +362,27 @@ namespace alpaka
                 std::apply(
                     [&](remove_restrict_t<std::decay_t<TArgs>> const&... args)
                     {
-                        kernelName<<<
-                            gridDim,
-                            blockDim,
-                            static_cast<std::size_t>(blockSharedMemDynSizeBytes),
-                            queue.getNativeHandle()>>>(threadElemExtent, task.m_kernelFnObj, args...);
+                        // checks whether to launch cooperative or non-cooperative kernel
+                        if constexpr(TCooperative)
+                        {
+                            void const* kernelArgs[] = {&threadElemExtent, &task.m_kernelFnObj, &args...};
+
+                            TApi::launchCooperativeKernel(
+                                reinterpret_cast<void*>(kernelName),
```

Comment:
Can you check if the cast can be removed, after implementing https://github.com/alpaka-group/alpaka/pull/2307/files#r1714986927 ?

Reply:
No, you now have to do the same cast inside of launchCooperativeKernel